### PR TITLE
Add scripts for converting memo

### PR DESCRIPTION
**Pull request by TTTTao725**
Prerequisite: # Make sure you have git-lfs installed (https://git-lfs.com)
git lfs install
git clone https://huggingface.co/datasets/MiMe-MeMo/Corpus-v1.1

### CODE REVIEW

File: data-processing/scripts/memo/convert_memo_to_jsonlgz.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    "text": "foo",           # MANDATORY: textual content of the document
+    "source": "...",         # MANDATORY: source of the data, such as peS2o, common-crawl, etc.
+    "added": "...",          # OPTIONAL: timestamp we acquired this data (time file was created), specified as YYYY-MM-DDTHH:MM:SS.TIMEZONE e.g 2021-04-13T12:52:46.000Z
+    "created": "..."         # OPTIONAL: timestamp when orig document was created (best-guess if not available), should be specified as a range; "YYYY-MM-DDTHH:MM:SS.TIMEZONE, YYYY-MM-DDTHH:MM:SS.TIMEZONE"
+    "metadata": {            # OPTIONAL: source-specific metadata
+										"sub-source": "...", # OPTIONAL: E.g. "newspaper_ocr"
+										...
+								}        
+}
+'''
```
Comment (KennethEnevoldsen) [inline data-processing/scripts/memo/convert_memo_to_jsonlgz.py:24]:
```suggestion
'''
.txt (text) & .csv(metadata) -> json.gz:

{
    "id": "...",             # MANDATORY: source-specific identifier
    "text": "foo",           # MANDATORY: textual content of the document
    "source": "...",         # MANDATORY: source of the data, such as peS2o, common-crawl, etc.
    "added": "...",          # OPTIONAL: timestamp we acquired this data (time file was created), specified as YYYY-MM-DDTHH:MM:SS.TIMEZONE e.g 2021-04-13T12:52:46.000Z
    "created": "..."         # OPTIONAL: timestamp when orig document was created (best-guess if not available), should be specified as a range; "YYYY-MM-DDTHH:MM:SS.TIMEZONE, YYYY-MM-DDTHH:MM:SS.TIMEZONE"
    "metadata": {            # OPTIONAL: source-specific metadata
										"sub-source": "...", # OPTIONAL: E.g. "newspaper_ocr"
										...
								}        
}
'''


import pandas as pd
import json
import gzip
import os
from tqdm import tqdm
import datetime
import nltk
from nltk.tokenize import word_tokenize
```

assuming it is a docstring I would also write it out a bit. I.e. "This script converts ..."

**Review by peterbjorgensen — CHANGES_REQUESTED** [review]
Can you set the created to a range instead of unknown if the year is unknown? I guess we know the range with an accuracy of +/- 100 years or something.

Otherwise it looks good to me.

**Review by KennethEnevoldsen — CHANGES_REQUESTED** [review]
There are a few changes I would add otherwise looks good.

**Comment by TTTTao725** [comment]
Hi guys, I changed the commit to make it more readable, please check it out :)

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good

**Comment by TTTTao725** [comment]
Hi guys, please check out the scripts for infomedia (converting and filtering)

unfiltered, filtered datasets and the list of empty files in the original dataset, all of them are here:
dfm-data/pre-training/danews2.0

File: e -i HEAD~2
Comment (KennethEnevoldsen) [inline e -i HEAD~2:1]: seems like this file should be deleted.

File: data-processing/scripts/infomedia/convert_infomedia_to_jsonlgz.py
Comment (KennethEnevoldsen) [inline data-processing/scripts/infomedia/convert_infomedia_to_jsonlgz.py:1]: Next time I would have done this in another PR, but no it is no biggie.

File: data-processing/scripts/infomedia/convert_infomedia_to_jsonlgz.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    """Create a formatted string representing a time range starting from `publish_date`."""
+    start_date = datetime.strptime(publish_date, "%Y-%m-%dT%H:%M:%SZ")
+    end_date = start_date + timedelta(days=365)
+    return f"{start_date.strftime('%Y-%m-%dT%H:%M:%SZ')}, {end_date.strftime('%Y-%m-%dT%H:%M:%SZ')}"
+
+def remove_html_tags(text):
+    """Remove HTML tags from a string."""
+    html_tag_pattern = re.compile('<.*?>')
+    clean_text = re.sub(html_tag_pattern, '', text)
+    return clean_text
```
Comment (KennethEnevoldsen) [inline data-processing/scripts/infomedia/convert_infomedia_to_jsonlgz.py:39]: @jankounchained how did you do the html filtering when you did it?

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good. Please remove the odd additional file.

**Comment by KennethEnevoldsen** [comment]
@TTTTao725 what is the number of tokens without information in infomedia?

File: e -i HEAD~2
Comment (TTTTao725) [reply e -i HEAD~2:1]: 👌🏼

**Comment by TTTTao725** [comment]
6578205491 / 6753608085 tokens after filtering

**Comment by KennethEnevoldsen** [comment]
Why the slash?

**Comment by TTTTao725** [comment]
that's just the number before filtering :)

**Comment by KennethEnevoldsen** [comment]
so Infomedia is ~175m tokens

**Comment by TTTTao725** [comment]
Sorry for the confusion, I like to write it like a processing bar 😹

**Comment by TTTTao725** [comment]
Oh I forgot to say, it does not include titles, summaries, etc, just text. So basically I just sum over the 'WordCount' instead of using a tokenizer which would be super slow

**Comment by TTTTao725** [comment]
So it's actually more than 6578205491 :)

**Comment by KennethEnevoldsen** [comment]
Ah so information is ~175m tokens

**Comment by TTTTao725** [comment]
I can't merge it Kenneth, merging is block
image

**Comment by KennethEnevoldsen** [comment]
just check the box and merge

File: data-processing/scripts/infomedia/convert_infomedia_to_jsonlgz.py (hunk 2 above)
Comment (jankounchained) [reply data-processing/scripts/infomedia/convert_infomedia_to_jsonlgz.py:39]: @KennethEnevoldsen @TTTTao725 sorry missed this!
the regex pattern looks good.

if you want to be extra careful, then you control for two edge cases:
- add a multiline flag in case there is a long html tag between mulitple lines
- replace html tags with a single shitespace in case that this happens: `word<br /></p><p id=\"p1\">word`
- take care of excess whitespace with another function

so something like this:
```
def remove_html_tags(text: str) -> str:
    """Remove HTML tags from a string."""
    html_tag_pattern = re.compile('<.*?>', flags=re.MULTILINE)
    clean_text = re.sub(html_tag_pattern, " ", text)
    return clean_text

def remove_whitespace(text: str) -> str:
    """remove excess whitespace from text fields"""
    pat_ws = re.compile(pattern=r"\s\s+", flags=re.MULTILINE)
    clean_text = re.sub(html_tag_pattern, " ", text)
    return clean_text
```
